Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeapon.cs

```diff
@@ -27,33 +28,36 @@ public void DecrementCooldown(float deltaTime)
             _currentCooldown -= _currentCooldown > 0 ? deltaTime : _currentCooldown;
         }
 
-        public void AttackManyIfPossible(Vector3 playerPosition, Vector3 playerForwardDirection)
+        public bool AttackManyIfPossible(Vector3 playerPosition, Vector3 playerForwardDirection)
         {
             if (!CanMelee)
-                return;
+                return false;
             _currentCooldown += _meleeProperties.AttackCoolDown;
             
-            var hits = _meleeProperties.OverlapCapsule(playerPosition, playerForwardDirection, ref _enemyBuffer);
-            if (hits <= 0) 
-                return;
-            
-            for(int i = 0; i < hits; i++)
-            {
-                DamageEnemy(_enemyBuffer[i]);
-            }
+            _hits = _meleeProperties.OverlapCapsule(playerPosition, playerForwardDirection, ref _enemyBuffer);
+            return true;
         }
         
-        public void AttackOneIfPossible(Vector3 playerPosition, Vector3 playerForwardDirection)
+        public bool AttackOneIfPossible(Vector3 playerPosition, Vector3 playerForwardDirection)
         {
             if (!CanMelee)
-                return;
+                return false;
             _currentCooldown += _meleeProperties.AttackCoolDown;
             
             var hits = _meleeProperties.OverlapCapsule(playerPosition, playerForwardDirection, ref _enemyBuffer);
-            if (hits <= 0) 
+            if (hits > 1)
+                hits = 1; //ensure we only damage first enemy
```

Comment:
should this be `_hits` or am I missing something?

Reply:
You're totally right, good catch!